### ISSUE TITLE
Detecting that the monitored Sentinel node or cluster is down

### ISSUE THREAD

**Issue by slovdahl** [leominov/redis_sentinel_exporter#16]
At the moment, I can't seem to find a way to e.g. create an alert for when a Sentinel instance is down. Many Prometheus exporters usually have a `foo_up` time series that returns 1 as long as the monitored node is up and running, but this exporter doesn't seem to have any such time series. Am I missing something obvious?

It also seems that the previous state of the metrics continues to be returned if the Sentinel process is stopped while the exporter is running. Is this intentional?

**Comment by leominov** [comment]
Hello, @slovdahl, just try `redis_sentinel_exporter_last_scrape_error`, also common-used metric name to indication any problems with data source.

**Comment by slovdahl** [comment]
Thanks! :+1: That probably works fine as a replacement for `foo_up`.

What about this:

> It also seems that the previous state of the metrics continues to be returned if the Sentinel process is stopped while the exporter is running. Is this intentional?

It feels kind of strange to have the same data scraped even when the node is down. If nothing else, it may cause Grafana graphs to look strange. :thinking: What are your thoughts about it?

**Comment by leominov** [comment]
Looks like a bug, I'll check!

**Comment by perlun** [comment]
> It feels kind of strange to have the same data scraped even when the node is down.

I agree about this. I get the feeling that this part in `exporter.go` would perhaps make sense to edit a bit:

```go
	if err != nil {
		errorCount++
	} else {
		sentinelInfo := ParseInfo(infoRaw, metricRequiredKeys, true)
		e.setMetrics(sentinelInfo)
	}
```

By either:

1. Calling `e.setMetrics(sentinelInfo)` with another info object in case parsing failed, which indicates the failure, _or_
2. Doing some kind of `e.clearMetrics()` before the scraping altogether (but this could mean that we sometimes return an empty dataset, which would perhaps be just as bad)

**Comment by leominov** [comment]
Needs some `metric.Reset()` magic :)

**Comment by leominov** [comment]
Fixed in 1.8.1, @slovdahl can you test it? :)
